File: Chapter08/recipe-07/cxx-example/src/io/io.cpp
#include "io.hpp"

#include <iostream>
#include <vector>

void print_row(const std::vector<int> row) {
  for (int i = 0; i < row.size(); i++) {
    if (row[i] == 1) {
      std::cout << "*";
    } else {
      std::cout << " ";
    }
  }
  std::cout << std::endl;
}

File: chapter-08/recipe-09/fortran-example/src/io/io.f90
module io

  implicit none
  public print_row
  private

contains

  subroutine print_row(row)
    integer, intent(in) :: row(:)
    character(size(row)) :: line
    integer :: i

    do i = 1, size(row)
      if (row(i) == 1) then
        line(i:i) = '*'
      else
        line(i:i) = ' '
      end if
    end do

    print *, line
  end subroutine

end module
